executor/aggfuncs/func_variance.go

```diff
@@ -0,0 +1,431 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aggfuncs
+
+import (
+	"math"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/chunk"
+	"github.com/pingcap/tidb/util/set"
+)
+
+type varianceType int
+
+const (
+	varianceNone varianceType = iota
+	varianceVarPop
+	varianceVarSamp
+	varianceStdPop
+	varianceStdSamp
+)
+
+// "baseVarianceDecimal" is wrapped by:
+// - "varianceOriginal4Decimal"
+// - "variancePartial4Decimal"
+type baseVarianceDecimal struct {
+	baseAggFunc
+	varianceType
+}
+
+type partialResult4VarianceDecimal struct {
+	count    int64
+	sum      types.MyDecimal
+	variance types.MyDecimal
+}
+
+func (p *partialResult4VarianceDecimal) merge(count int64, sum, variance types.MyDecimal) error {
+	if p.count == 0 {
+		p.count, p.sum, p.variance = count, sum, variance
+	} else {
+		variance, sum, err := types.CalculateMergeDecimal(p.count, count, &p.sum, &sum, &p.variance, &variance)
```

Comment:
After function `calculateVariance()` being removed, the only caller of function `CalculateMergeDecimal()` is this function. Considered leaving a function which merges the two partial result of variance in the package `types` is weird, we can remove the function `CalculateMergeDecimal()` as well, put the code into this function.

Reply:
Maybe it's better to put it in `types/help.go`. Because it will be helpful in the future for other extensions. And also for convenient to test it.

Reply:
we can move it to `types/help.go` if we need it in the future.

Reply:
Actually...My main propose is to test it conveniently.   - -|